service/controller/rolebindingtemplate/resource/rolebinding/create.go

```diff
@@ -38,9 +38,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		roleBinding = cleanSubjects(roleBinding, ns)
 		if len(roleBinding.Subjects) > 0 {
 			if err = rbac.CreateOrUpdateRoleBinding(r, ctx, ns, roleBinding); err != nil {
-				r.logger.Debugf(ctx, "Could not apply roleBinding %s to namespace %s due to error %v", roleBinding.Name, ns, err)
-				continue
+				r.logger.Errorf(ctx, err, "Failed to apply roleBinding %s to namespace %s", roleBinding.Name, ns)
+				return microerror.Maskf(roleBindingCreationFailedError, "failed to apply role binding to namespace %s: %v", ns, err)
```

Comment:
Returning an error here will result in unintentionally skipping other namespaces in the for loop.
For example, if we have the following namespaces:
```
a-namespace
a-second-namespace
a-very-bad-namespace    <-- some error occurs handling this namespace. This function returns
this-other-namespace       <-- this namespace is never handled
```

Reply:
Here you could actually either emit an event or add a status condition